packages/api/tests/lib/test-granule-delete.js

```diff
@@ -0,0 +1,503 @@
+
+  // create PG granule
+  const fakePGGranule = fakeGranuleRecordFactory(
+    {
+      granule_id: granuleId,
+      status: 'failed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+    }
+  );
+  fakePGGranule.published = false;
+  const [granuleCumulusId] = await granulePgModel.create(t.context.knex, fakePGGranule);
+
+  const newPgGranule = await granulePgModel.get(t.context.knex, { cumulus_id: granuleCumulusId });
+  const newDynamoGranule = await granuleModel.get({ granuleId: newGranule.granuleId });
+
+  await deleteGranuleAndFiles({
+    knex: t.context.knex,
+    dynamoGranule: newDynamoGranule,
+    pgGranule: newPgGranule,
+  });
+
+  // Check Dynamo and RDS. The granule should have been removed from both.
+  t.false(
+    await granuleModel.exists({ granuleId: newDynamoGranule.granuleId })
+  );
+
+  t.false(
+    await granulePgModel.exists(t.context.knex, { granule_id: newPgGranule.granule_id })
+  );
+});
+
+test.serial('deleteGranuleAndFiles() will not delete a granule or its S3 files if the PG file delete fails', async (t) => {
+  const {
+    newPgGranule,
+    newDynamoGranule,
+    files,
+    s3Buckets,
+  } = await createGranuleAndFiles(
+    t.context.knex,
+    t.context.collectionCumulusId,
+    false
+  );
+
+  const mockFileModel = {
+    tableName: 'files',
+    delete: () => {
+      throw new Error('Delete failed');
+    },
+  };
+
+  await t.throwsAsync(
+    deleteGranuleAndFiles({
+      knex: t.context.knex,
+      dynamoGranule: newDynamoGranule,
+      pgGranule: newPgGranule,
+      filePgModel: mockFileModel,
+      granuleModelClient: granuleModel,
+    }),
+    { instanceOf: Error }
+  );
+
+  // granule should still exist in Dynamo and PG
+  t.true(await granulePgModel.exists(t.context.knex, { granule_id: newPgGranule.granule_id }));
+  t.true(await granuleModel.exists({ granuleId: newDynamoGranule.granuleId }));
+
+  // Verify files still exist in S3 and PG
+  await Promise.all(
+    files.map(async (file) => {
+      t.true(await s3ObjectExists({ Bucket: file.bucket, Key: file.key }));
+      t.true(await filePgModel.exists(t.context.knex, { bucket: file.bucket, key: file.key }));
+    })
+  );
+
+  t.teardown(() => deleteS3Buckets([
+    s3Buckets.protected.name,
+    s3Buckets.public.name,
+  ]));
+});
+
+test.serial('deleteGranuleAndFiles() will not delete PG or S3 Files if the PG Granule delete fails', async (t) => {
+  const {
+    newPgGranule,
+    newDynamoGranule,
+    files,
+    s3Buckets,
+  } = await createGranuleAndFiles(
+    t.context.knex,
+    t.context.collectionCumulusId,
+    false
+  );
+
+  const mockGranuleModel = {
+    tableName: 'granules',
+    delete: () => {
+      throw new Error('Delete failed');
+    },
+  };
+
+  await t.throwsAsync(
+    deleteGranuleAndFiles({
+      knex: t.context.knex,
+      dynamoGranule: newDynamoGranule,
+      pgGranule: newPgGranule,
+      granulePgModel: mockGranuleModel,
+      granuleModelClient: granuleModel,
```

Comment:
We should be able to remove this

Reply:
👍 https://github.com/nasa/cumulus/pull/2057/commits/35dc237c18f7d7aa47dd079f17b7703d5129df1a